### DISCUSSION TITLE
Performance issues with Tween.StopAll

### DISCUSSION THREAD

**Discussion by myangel0258** [KyryloKuzyk/PrimeTween#128] [FAQ] [answered]
We are going to use PrimeTween to replace DoTween, but there is a performance issue that needs to be confirmed. There are a large number of GameObjects in the scene, and each GameObject plays a large number of tween animations at the same time. I need to call tween.stopAll(target) to stop all tween animations on a certain GameObject. In this usage scenario, I would like to know the performance comparison between PrimeTween's **Tween.stopAll(target)** and Dotween's **DOTween.Kill(target)**. Look forward to your reply and suggestions.

**Comment by KyryloKuzyk** [comment, marked as answer]
Hey! 

I would like to start by saying that both PrimeTween's `Tween.StopAll(target)` and `DOTween.Kill(targetOrId)` don't support stopping all animations on a particular GameObject hierarchy. The GameObject type is rarely an animation target unless you're doing something like `Tween.Custom(gameObject, ...)`. So, if you have a position animation that has a Transform target, it would not be possible to stop it by passing GameObject target.

In regards to performance, both PrimeTween's `Tween.StopAll(target)` and `DOTween.Kill(targetOrId)` are not the best choices if you call them frequently on many objects because they iterate over all running animations. And if you do that for all running animations, you'll get O(n^2) complexity, which is extremely bad.

Instead of using `Tween.StopAll()`, save all references to all running animations and stop them directly via the reference. This would be the fastest way. Or if you run many animations on the same object at the same time, you can group all of them in one Sequence and store a reference to the sequence instead of storing each tween reference individually.

Please tell me if this helps.

**Reply by myangel0258** [reply]
Thank you very much for your reply. After we discovered the performance issue of Tween.stopAll, our approach coincides with your suggestion.
